examples/nvector/mpimanyvector/test_fnvector_mpimanyvector_mod.f90

```diff
@@ -25,37 +25,37 @@ module test_nvector_mpimanyvector
   implicit none
   include "mpif.h"
 
-  integer(c_int), parameter  :: nsubvecs = 2
-  integer(c_long), parameter :: N1       = 100        ! individual vector length
-  integer(c_long), parameter :: N2       = 200        ! individual vector length
-  integer(c_int),  parameter :: nv       = 3          ! length of vector arrays
-  integer(c_long), parameter :: N        = N1 + N2    ! overall manyvector length
-  integer(c_int), target     :: comm = MPI_COMM_WORLD ! default MPI communicator
-  integer(c_int)             :: nprocs                ! number of MPI processes
+  integer(kind=myindextype), parameter :: nsubvecs = 2
+  integer(kind=myindextype), parameter :: N1 = 100      ! individual vector length
+  integer(kind=myindextype), parameter :: N2 = 200      ! individual vector length
+  integer(kind=myindextype), parameter :: nv = 3        ! length of vector arrays
+  integer(kind=myindextype), parameter :: N = N1 + N2   ! overall manyvector length
+  integer(c_int), target :: comm = MPI_COMM_WORLD       ! default MPI communicator
+  integer(c_int) :: nprocs                              ! number of MPI processes
 
 contains
 
   integer function smoke_tests() result(ret)
     implicit none
 
-    integer(c_long)         :: lenrw(1), leniw(1)     ! real and int work space size
-    integer(c_long)         :: ival                   ! integer work value
-    real(c_double)          :: rval                   ! real work value
-    real(c_double)          :: x1data(N1), x2data(N2) ! vector data array
-    real(c_double), pointer :: xptr(:)                ! pointer to vector data array
-    real(c_double)          :: nvarr(nv)              ! array of nv constants to go with vector array
-    type(N_Vector), pointer :: x, y, z, tmp           ! N_Vectors
-    type(c_ptr)             :: subvecs                ! MPIManyVector subvectors
-    type(c_ptr)             :: xvecs, zvecs           ! C pointer to array of MPIManyVectors
+    integer(kind=myindextype) :: lenrw(1), leniw(1)     ! real and int work space size
+    integer(kind=myindextype) :: ival                   ! integer work value
+    real(c_double)            :: rval                   ! real work value
+    real(c_double)            :: x1data(N1), x2data(N2) ! vector data array
+    real(c_double), pointer   :: xptr(:)                ! pointer to vector data array
+    real(c_double)            :: nvarr(nv)              ! array of nv constants to go with vector array
+    type(N_Vector), pointer   :: x, y, z, tmp           ! N_Vectors
+    type(c_ptr)               :: subvecs                ! MPIManyVector subvectors
+    type(c_ptr)               :: xvecs, zvecs           ! C pointer to array of MPIManyVectors
 
     !===== Setup ====
     subvecs = FN_VNewVectorArray(nsubvecs, sunctx)
     tmp  => FN_VMake_Serial(N1, x1data, sunctx)
-    call FN_VSetVecAtIndexVectorArray(subvecs, 0, tmp)
+    call FN_VSetVecAtIndexVectorArray(subvecs, 0_myindextype, tmp)
```

Comment:
I guess we were inconsistent with how `VectroArray`s are index (`int`) and how subvectors are indexed (`sunindextype`). These should probably be the same and I guess it's easier to promote `int` to `sunindextype` even if we'd probably never need even max int numbers of subvectors.